### ISSUE TITLE
checkstyle and googlejavaformat conflicting rules

### ISSUE THREAD

**Issue by z035079** [checkstyle/checkstyle#3591]
I am getting indentation checkstyle warning message for annotation class. 
Attached is the java class and XML config files. 
I am getting  `')' have incorrect indentation level 2, expected level should be 4` message. 
This contradicts with GoogleJavaFormat rule. Please advise.

```
$ javac PackageDimensions.java 

/var/tmp$ cat PackageDimensions.java 
public class PackageDimensions {

  @SuppressWarnings(
    {"unchecked", "deprecation"}
  ) // violation
  @Deprecated
  private int height;

  @SuppressWarnings(
    {"unchecked", "deprecation"}
  ) // no voilation
  private int height2;

  void foo () {
    indentationOfBrace(
      "as",
      "adsf"
    ); // no violation
  }

  void indentationOfBrace(String v, String v2) {}

}

rivanov@p5510:/var/tmp$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="2"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-7.3-all.jar -c config.xml PackageDimensions.java
Starting audit...
[ERROR] /var/tmp/PackageDimensions.java:5: ')' have incorrect indentation level 2, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

original code looks like:
```
public class PackageDimensions {

  @Digits(
    integer = 8,
    fraction = 2,
    message =
        "Must be a numeric value with a maximum of 8 digits before decimal point and "
            + "2 digits after decimal."
  )
  @DecimalMin(value = "0.00", message = "The decimal value can not be less than 0.00 digit.")
  private BigDecimal height;

}
```

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/39598179-checkstyle-and-googlejavaformat-conflicting-rules?utm_campaign=plugin&utm_content=tracker%2F500528&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F500528&utm_medium=issues&utm_source=github).

**Comment by romani** [comment]
please report issue as it was requested in template - https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE.md

It is mandatory, all requested details do help us a lot.

**Comment by z035079** [comment]
Sorry about that; here is the code, xml config and the output of the checkstyle CLU

$ cat ~/item-nebula/src/main/java/com/tgt/enterprise/item/dto/PackageDimensions.java
package com.tgt.enterprise.item.dto;

import com.tgt.enterprise.item.constants.ItemConstants;
import com.tgt.enterprise.item.interfaces.CheckEnum;
import com.tgt.enterprise.item.interfaces.CheckField;
import java.math.BigDecimal;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.Digits;
import org.hibernate.validator.constraints.Length;

@CheckField
public class PackageDimensions {

  @Digits(
    integer = 8,
    fraction = 2,
    message =
        "Must be a numeric value with a maximum of 8 digits before decimal point and "
            + "2 digits after decimal."
  )
  @DecimalMin(value = "0.00", message = "The decimal value can not be less than 0.00 digit.")
  private BigDecimal height;

  @Digits(
    integer = 8,
    fraction = 2,
    message =
        "Must be a numeric value with a maximum of 8 digits before decimal point and "
            + "2 digits after decimal."
  )
  @DecimalMin(value = "0.00", message = "The decimal value can not be less than 0.00 digit.")
  private BigDecimal depth;

  @Digits(
    integer = 8,
    fraction = 2,
    message =
        "Must be a numeric value with a maximum of 8 digits before decimal point and "
            + "2 digits after decimal."
  )
  @DecimalMin(value = "0.00", message = "The decimal value can not be less than 0.00 digit.")
  private BigDecimal width;

  @Length(max = 10, message = "Unit of measure cannot exceed 10 characters")
  @CheckEnum(enumClass = ItemConstants.PackageDimensionUnitOfMeasure.class, ignoreCase = true)
  private String dimensionUnitOfMeasure;

  @Digits(
    integer = 8,
    fraction = 2,
    message =
        "Must be a numeric value with a maximum of 8 digits before decimal point and "
            + "2 digits after decimal."
  )
  @DecimalMin(value = "0.00", message = "The decimal value can not be less than 0.00 digit.")
  private BigDecimal weight;

  @Length(max = 10, message = "Unit of measure cannot exceed 10 characters")
  @CheckEnum(enumClass = ItemConstants.WeightUnitOfMeasure.class, ignoreCase = true)
  private String weightUnitOfMeasure;

  public PackageDimensions() {
    //Empty Constructor
  }

  public BigDecimal getHeight() {
    return height;
  }

  public void setHeight(final BigDecimal height) {
    this.height = height;
  }

  public BigDecimal getDepth() {
    return depth;
  }

  public void setDepth(final BigDecimal depth) {
    this.depth = depth;
  }

  public BigDecimal getWidth() {
    return width;
  }

  public void setWidth(final BigDecimal width) {
    this.width = width;
  }

  public String getDimensionUnitOfMeasure() {
    return dimensionUnitOfMeasure;
  }

  public void setDimensionUnitOfMeasure(final String dimensionUnitOfMeasure) {
    this.dimensionUnitOfMeasure = dimensionUnitOfMeasure;
  }

  public String getWeightUnitOfMeasure() {
    return weightUnitOfMeasure;
  }

  public void setWeightUnitOfMeasure(final String weightUnitOfMeasure) {
    this.weightUnitOfMeasure = weightUnitOfMeasure;
  }

  public BigDecimal getWeight() {
    return weight;
  }

  public void setWeight(final BigDecimal weight) {
    this.weight = weight;
  }

  public PackageDimensions(
      final BigDecimal height,
      final BigDecimal depth,
      final BigDecimal width,
      final String dimensionUnitOfMeasure,
      final BigDecimal weight,
      final String weightUnitOfMeasure) {
    this.height = height;
    this.depth = depth;
    this.width = width;
    this.dimensionUnitOfMeasure = dimensionUnitOfMeasure;
    this.weight = weight;
    this.weightUnitOfMeasure = weightUnitOfMeasure;
  }
}

$ cat ~/item-nebula/config/checkstyle/checkstyle.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">



<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <property name="severity" value="warning"/>

    <property name="fileExtensions" value="java, properties, xml"/>
    
    
    <module name="FileTabCharacter">
        <property name="eachLine" value="true"/>
    </module>

    <module name="TreeWalker">
        <module name="OuterTypeFilename"/>
        <module name="IllegalTokenText">
            <property name="tokens" value="STRING_LITERAL, CHAR_LITERAL"/>
            <property name="format" value="\\u00(08|09|0(a|A)|0(c|C)|0(d|D)|22|27|5(C|c))|\\(0(10|11|12|14|15|42|47)|134)"/>
            <property name="message" value="Avoid using corresponding octal or Unicode escape."/>
        </module>
        <module name="AvoidEscapedUnicodeCharacters">
            <property name="allowEscapesForControlCharacters" value="true"/>
            <property name="allowByTailComment" value="true"/>
            <property name="allowNonPrintableEscapes" value="true"/>
        </module>
        <module name="LineLength">
            <property name="max" value="100"/>
            <property name="ignorePattern" value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
        </module>
        <module name="AvoidStarImport"/>
        <module name="OneTopLevelClass"/>
        <module name="NoLineWrap"/>
        <module name="EmptyBlock">
            <property name="option" value="TEXT"/>
            <property name="tokens" value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH"/>
        </module>
        <module name="NeedBraces"/>
        <module name="LeftCurly">
            <property name="maxLineLength" value="100"/>
        </module>
        <module name="RightCurly"/>
        <module name="RightCurly">
            <property name="option" value="alone"/>
            <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
        </module>
        <module name="WhitespaceAround">
            <property name="allowEmptyConstructors" value="true"/>
            <property name="allowEmptyMethods" value="true"/>
            <property name="allowEmptyTypes" value="true"/>
            <property name="allowEmptyLoops" value="true"/>
            <message key="ws.notFollowed"
                     value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
            <message key="ws.notPreceded"
                     value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
        </module>
        <module name="OneStatementPerLine"/>
        <module name="MultipleVariableDeclarations"/>
        <module name="ArrayTypeStyle"/>
        <module name="MissingSwitchDefault"/>
        <module name="FallThrough"/>
        <module name="UpperEll"/>
        <module name="ModifierOrder"/>
        <module name="EmptyLineSeparator">
            <property name="allowNoEmptyLineBetweenFields" value="true"/>
        </module>
        <module name="SeparatorWrap">
            <property name="tokens" value="DOT"/>
            <property name="option" value="nl"/>
        </module>
        <module name="SeparatorWrap">
            <property name="tokens" value="COMMA"/>
            <property name="option" value="EOL"/>
        </module>
        <module name="PackageName">
            <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
            <message key="name.invalidPattern"
                     value="Package name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="TypeName">
            <message key="name.invalidPattern"
                     value="Type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MemberName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern"
                     value="Member name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="ParameterName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern"
                     value="Parameter name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="CatchParameterName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern"
                     value="Catch parameter name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="LocalVariableName">
            <property name="tokens" value="VARIABLE_DEF"/>
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <property name="allowOneCharVarInForLoop" value="true"/>
            <message key="name.invalidPattern"
                     value="Local variable name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="ClassTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Class type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MethodTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Method type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="InterfaceTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Interface type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="NoFinalizer"/>
        <module name="GenericWhitespace">
            <message key="ws.followed"
                     value="GenericWhitespace ''{0}'' is followed by whitespace."/>
            <message key="ws.preceded"
                     value="GenericWhitespace ''{0}'' is preceded with whitespace."/>
            <message key="ws.illegalFollow"
                     value="GenericWhitespace ''{0}'' should followed by whitespace."/>
            <message key="ws.notPreceded"
                     value="GenericWhitespace ''{0}'' is not preceded with whitespace."/>
        </module>
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="2"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
        <module name="AbbreviationAsWordInName">
            <property name="ignoreFinal" value="false"/>
            <property name="allowedAbbreviationLength" value="1"/>
        </module>
        <module name="OverloadMethodsDeclarationOrder"/>
        <module name="VariableDeclarationUsageDistance"/>
        <module name="CustomImportOrder">
            <property name="specialImportsRegExp" value="com.google"/>
            <property name="sortImportsInGroupAlphabetically" value="true"/>
        </module>
        <module name="MethodParamPad"/>
        <module name="OperatorWrap">
            <property name="option" value="NL"/>
            <property name="tokens" value="BAND, BOR, BSR, BXOR, DIV, EQUAL, GE, GT, LAND, LE, LITERAL_INSTANCEOF, LOR, LT, MINUS, MOD, NOT_EQUAL, PLUS, QUESTION, SL, SR, STAR "/>
        </module>
        <module name="AnnotationLocation">
            <property name="tokens" value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF"/>
        </module>
        <module name="AnnotationLocation">
            <property name="tokens" value="VARIABLE_DEF"/>
            <property name="allowSamelineMultipleAnnotations" value="true"/>
        </module>
        <module name="NonEmptyAtclauseDescription"/>
        <module name="JavadocTagContinuationIndentation"/>
        <module name="SummaryJavadoc">
            <property name="forbiddenSummaryFragments" value="^@return the *|^This method returns |^A [{]@code [a-zA-Z0-9]+[}]( is a )"/>
        </module>
        <module name="JavadocParagraph"/>
        <module name="AtclauseOrder">
            <property name="tagOrder" value="@param, @return, @throws, @deprecated"/>
            <property name="target" value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>
        </module>
        <module name="MethodName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9_]*$"/>
            <message key="name.invalidPattern"
                     value="Method name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="SingleLineJavadoc">
            <property name="ignoreInlineTags" value="false"/>
        </module>
        <module name="EmptyCatchBlock">
            <property name="exceptionVariableName" value="expected"/>
        </module>
        <module name="CommentsIndentation"/>
    </module>

$ java -jar checkstyle-7.3-all.jar -c ~/item-nebula/config/checkstyle/checkstyle.xml ~/item-nebula/src/main/java/com/tgt/enterprise/item/dto/PackageDimensions.java
Starting audit...
[WARN] /Users/z035079/item-nebula/src/main/java/com/tgt/enterprise/item/dto/PackageDimensions.java:20: ')' have incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] /Users/z035079/item-nebula/src/main/java/com/tgt/enterprise/item/dto/PackageDimensions.java:30: ')' have incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] /Users/z035079/item-nebula/src/main/java/com/tgt/enterprise/item/dto/PackageDimensions.java:40: ')' have incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] /Users/z035079/item-nebula/src/main/java/com/tgt/enterprise/item/dto/PackageDimensions.java:54: ')' have incorrect indentation level 2, expected level should be 4. [Indentation]
Audit done.

**Comment by romani** [comment]
@z035079 , what is  GoogleJavaFormat ?

**Comment by bamapookie** [comment]
I am having this same issue.  Looking at your code above, and confirming by editing my own code, the violation happens because the closing parenthesis is immediately followed by another annotation.  The violation does not occur when the closing parenthesis is followed by the variable declaration.  In my own case, it was a class annotation, and thus should have been at level 0, but checkstyle was reporting that the parenthesis should be at level 4.  Once I reordered the annotations to put the closing parenthesis before the class declaration, the error went away.

**Comment by jochenvdv** [comment]
@romani [google-java-format](https://github.com/google/google-java-format)

**Comment by romani** [comment]
how can we trust this tool ? 
Look at amount of issues, we could trust it only after it become completed.
We are probably in the same or different illusion/interpretation of what google style paper mean in some cases.

we are participating in clarification process of Google style paper, some points in google style nobody follow.
Non of googlers follow this format even it exists :), each group have its own but mostly similar style.

google style was added to checkstyle only to show that checkstyle can do this and substitute forgotten and outdated Sun style.

**Comment by bamapookie** [comment]
Regardless of whether the code matches the output of the tool, Checkstyle is exhibiting a bug here.  The following code reports an error:
```
@RightParenBug(
        "Check out that warning on the next line"
) // Error on this line: ')' have incorrect indentation level 0, expected level should be 4. (3:0)
@RightParenNoBug(
        "Check out that not-warning on the next line"
) // But no bug here.  This is what is expected.
public class CheckstyleRightParenBug {}
```

With the following checkstyle.xml:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation" />
    </module>
</module>
```

**Comment by romani** [comment]
@bamapookie , please review https://github.com/checkstyle/checkstyle/issues?q=is%3Aopen+is%3Aissue+label%3Aindentaion and report issue if it snot reported. Everyone is welcome to contribute.

> google-java-format

If smb wants to do comparison and deep analysis .... please do.

**Comment by bamapookie** [comment]
@romani thanks.  Will do.

**Comment by bamapookie** [comment]
Opened issue #3733.  Fixed in pull request #3737.

**Comment by rnveach** [comment]
I am closing this issue as it is a duplicate of #3733 and with it's fix, there are no more violations:
````
$ cat TestClass.java
public class PackageDimensions {

  @SuppressWarnings(
    {"unchecked", "deprecation"}
  ) // violation
  @Deprecated
  private int height;

  @SuppressWarnings(
    {"unchecked", "deprecation"}
  ) // no voilation
  private int height2;

  void foo () {
    indentationOfBrace(
      "as",
      "adsf"
    ); // no violation
  }

  void indentationOfBrace(String v, String v2) {}

}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="2"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-nightly-2017-02-15-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````
